### PR TITLE
Issue#24 current year coherency

### PR DESCRIPTION
**Pull request by SamuelVch98**
The current year was defined in two different ways and this posed a problem of consistency. I defined the current year statically using the ‘Configuration’ table.

### CODE REVIEW

File: util.py
Comment (anthonygego) [inline util.py]: There is no need to catch an exception if you don't perform any handling. Just leave the code as is, and do exception handling at the higher level.

File: templates/layout.html
Comment (anthonygego) [inline templates/layout.html]: It could be nice to have the current academic year shown somewhere.

File: config.py
Comment (anthonygego) [inline config.py]: The entry can probably be inserted automatically if needed to improve user friendlyness. Besides, `make_response` is not a very elegant way to return feedback to the user.

File: config.py
Comment (anthonygego) [inline config.py]: @luciledierckx Using this implementation we'll always be going in one way. Is there any case for which we'd like to revert and go to the past ?

File: course.py
Comment (anthonygego) [inline course.py]: Please avoid putting required arguments in request args: put them in the route, there are automatically sanitized.

File: config.py
Comment (anthonygego) [inline config.py]: Maybe you can return a feedback to the user in this case.

**Review by anthonygego — CHANGES_REQUESTED** [review]

File: config.py
Comment (luciledierckx) [reply config.py]: The only reason we would like to revert would be that we mistakenly call the next_year functionality more than once a year. As this action is available for all admins, maybe it would be safer to have an option to go back in case one of them mistakenly calls it?

File: config.py
Comment (anthonygego) [inline config.py]: Maybe we can log the error or at least give an insight in the feedback of what's gone wrong. There is almost no chance that "trying again" will work.

File: config.py
Comment (anthonygego) [inline config.py]: Same remark here.

File: config.py
Comment (anthonygego) [inline config.py]: `is_new_year` => `new_year`. It is confusing as it is not a boolean.

File: templates/layout.html
Comment (anthonygego) [inline templates/layout.html]:
```suggestion
                        <div class="navbar-text ms-0">
```

File: config.py
Comment (AlexandreDoneux) [inline config.py]: You can't delete a year if you already attributed it as being the current year. That doesn't solve the problem mentioned by Lucile. 

Moreover, we don't really want to delete a year, we just want to change the current year to a previous year. You **could** adapt the `/next_year` endpoint into a `/change_year` where you can specify a certain year. That way, you don't need a second endpoint to go to a previous year.

File: config.py
Comment (AlexandreDoneux) [inline config.py]: Not a helpful error message.

File: config.py
Comment (AlexandreDoneux) [inline config.py]: Not a helpful message.

File: config.py
Comment (AlexandreDoneux) [inline config.py]: You don't need that redirect as it is already present after your try-except

**Review by AlexandreDoneux — CHANGES_REQUESTED** [review]
Change some error messages and change the way you handle year "deletion"

**Comment by AlexandreDoneux** [comment]
Looks good to me, waiting for @anthonygego approval

**Review by anthonygego — APPROVED** [review]

**Pull request merged**
